parcels/field.py

```diff
@@ -826,6 +839,73 @@ def ccode_eval(self, var, t, z, y, x):
     def ccode_convert(self, _, z, y, x):
         return self.units.ccode_to_target(x, y, z)
 
+    #def find_in_block(self, chunksize, ti, zi, yi, xi):
+    #    #index = (ti, zi, yi, xi)
+    #    index = (ti, yi, xi)
+    #    block = tuple((int(index[i]/chunksize[i]) for i in range(len(index))))
+    #    local_index = tuple(index[i] - chunksize[i]*block[i] for i in range(len(index)))
+    #    return block, local_index
+
+    def get_block_id(self, block):
+        return np.ravel_multi_index(block, self.nchunks)
+
+    def get_block(self, bid):
+        return np.unravel_index(bid, self.nchunks)
+
+    def chunk_data(self):
+        if isinstance(self.data, da.core.Array):
+            if len(self.data.chunks[0]) > 1:
+                self.data = self.data.rechunk(sum(((self.grid.tdim,), self.data.chunksize[1:]), ()))
+            npartitions = self.data.npartitions
+            chunksize = self.data.chunksize
+            chunks = self.data.chunks
+        else:
+            npartitions = 1
+            chunksize = self.data.shape
+            chunks = tuple((t,) for t in chunksize)
+
+        self.nchunks = tuple(len(chunks[i]) for i in range(len(chunks)))
+        if len(self.data_chunks) == 0:
+            self.data_chunks = [None] * npartitions
+            self.c_data_chunks = [None] * npartitions
+
+        if len(self.grid.load_chunk) == 0:
+            self.grid.load_chunk = np.zeros(npartitions, dtype=c_int)
+        #print(self.grid.load_chunk)
+
+        # self.grid.load_chunk code:
+        # 0: not loaded
+        # 1: was asked to load
+        # 2: is loaded and was touched last C call
+        # 3: is loaded
+        if isinstance(self.data, da.core.Array):
+            for block_id in range(len(self.grid.load_chunk)):
+                if self.grid.load_chunk[block_id] == 1:
+                    #block, local_index = self.find_in_block(chunksize, 0, 0, 20, 40)
+                    #block_id = self.get_block_id(nchunks, block)
+                    block = self.get_block(block_id)
+                    self.data_chunks[block_id] = np.array(self.data.blocks[block])
```

Comment:
Can't you index `dask.array.blocks` by a scalar anyway to get this same behaviour?